orb-slot-ctrl/src/main.rs

```diff
@@ -1,204 +1,14 @@
-                        "unbootable" => orb_slot_ctrl::RootFsStatus::Unbootable,
-                        "3" => orb_slot_ctrl::RootFsStatus::Unbootable,
-                        _ => {
-                            println!("Invalid status provided. For a full list of available rootfs status run:");
-                            println!("slot-ctrl status --list");
-                            exit(1)
-                        }
-                    };
-                    if inactive {
-                        if let Err(e) = orb_slot_ctrl::set_rootfs_status(
-                            status,
-                            orb_slot_ctrl::get_inactive_slot()?,
-                        ) {
-                            check_running_as_root(e);
-                        }
-                    } else if let Err(e) =
-                        orb_slot_ctrl::set_current_rootfs_status(status)
-                    {
-                        check_running_as_root(e);
-                    }
-                }
-                StatusCommands::GetRetryCounter => {
-                    if inactive {
-                        println!(
-                            "{}",
-                            orb_slot_ctrl::get_retry_count(
-                                orb_slot_ctrl::get_inactive_slot()?
-                            )?
-                        );
-                    } else {
-                        println!("{}", orb_slot_ctrl::get_current_retry_count()?);
-                    }
-                }
-                StatusCommands::GetMaxRetryCounter => {
-                    println!("{}", orb_slot_ctrl::get_max_retry_count()?);
-                }
-                StatusCommands::ResetRetryCounter => {
-                    if inactive {
-                        if let Err(e) = orb_slot_ctrl::reset_retry_count_to_max(
-                            orb_slot_ctrl::get_inactive_slot()?,
-                        ) {
-                            check_running_as_root(e)
-                        }
-                    } else if let Err(e) =
-                        orb_slot_ctrl::reset_current_retry_count_to_max()
-                    {
-                        check_running_as_root(e)
-                    }
-                }
-                StatusCommands::ListStatusVariants => {
-                    println!("Available Rootfs status variants with their aliases):");
-                    println!("  Normal (normal, 0)");
-                    println!("  UpdateInProcess (updateinprocess, updinprocess, 1)");
-                    println!("  UpdateDone (updatedone, upddone, 2)");
-                    println!("  Unbootable (unbootable, 3)");
-                }
-            }
-        }
-        Commands::GitDescribe => {
-            println!("{}", BUILD_INFO.git.describe);
-        }
-    }
-    Ok(())
+
+    let db = EfiVarDb::from_rootfs("/")?;
+    let orb_slot_ctrl = OrbSlotCtrl::new(&db)?;
+
+    program::run(&orb_slot_ctrl, cli)
```

Comment:
Why do we move `program` into a new file?

Reply:
We can't import things from `main.rs`  on integration tests :(